src/libraries/System.Runtime.Numerics/src/System/Numerics/Decimal32.cs

```diff
@@ -0,0 +1,1699 @@
+
+        private const uint PositiveInfinityBits = 0x7800_0000;
+        private const uint NegativeInfinityBits = SignMask | PositiveInfinityBits;
+
+
+        // QNanBits
+        // Hex:                   0x7C00_0000
+        // Binary:                0111_1100_0000_0000_0000_0000_0000_0000
+        // Split into sections:   0 | 111_1100_0000 | 0000_0000_0000_0000_0000
+        // Section labels:        a | b             | c
+        //
+        // a. Sign bit (ignored for NaN).
+        // b. Combination field G0 through G10. G0-G4 == 11111 encodes NaN.
+        // c. Trailing significand. Can be used to encode a payload, to distinguish different NaNs.
+        // Note: Canonical NaN has G6-G10 as 0 and the encoding of the payload also canonical.
+
+        private const uint QNanBits = 0x7C00_0000; // TODO I used a "positive" NaN here, should it be negative?
+        private const uint SNanBits = 0x7E00_0000;
+
+
+        // MaxValueBits
+        // Hex:                   0x77F8_967F
+        // Binary:                0111_0111_1111_1000_1001_0110_0111_1111
+        // Split into sections:   0 | 11 | 1_0111_111 | 1_1000_1001_0110_0111_1111
+        //                        0 | 11 | 1011_1111  | [100]1_1000_1001_0110_0111_1111
+        // Section labels:        a | b  | c          | d
+        //
+        // a. Sign bit.
+        // b. G0 and G1 of the combination field, "11" indicates this version of encoding.
+        // c. Biased Exponent, which is q + 101. 1011_1111 == 191 == 90 + 101, so this is encoding an q of 90.
+        // d. Significand. Section b. indicates an implied prefix of [100]. [100]1_1000_1001_0110_0111_1111 == 9,999,999.
+        //
+        // Encoded value:         9,999,999 x 10^90
+
+        private const uint MaxValueBits = 0x77F8_967F;
+        private const uint MinValueBits = SignMask | MaxValueBits;
+
+        // PositiveOneBits Bits
+        // Hex:                   0x3280_0001
+        // Binary:                0011_0010_1000_0000_0000_0000_0000_0001
+        // Split into sections:   0 | 011_0010_1 | 000_0000_0000_0000_0000_0001
+        //                        0 | 0110_0101  | 000_0000_0000_0000_0000_0001
+        // Section labels:        a | b          | c
+        //
+        // a. Sign bit.
+        // b. Biased Exponent, which is q + 101. 0110_0101 == 101 == 0 + 101, so this is encoding an q of 0.
+        // c. Significand, set to 1.
+        //
+        // Encoded value:         1 x 10^0
+
+        private const uint PositiveOneBits = 0x3280_0001;
+        private const uint NegativeOneBits = SignMask | PositiveOneBits;
+
+        /*
+                private const uint EBits = 0; // TODO
+                private const uint PiBits = 0; // TODO
+                private const uint TauBits = 0; // TODO*/
+
+        internal readonly uint _value; // TODO: Single places this at the top, Half places it here. Also, Single has this as private, Half has it as internal. What do we want?
```

Comment:
I personally prefer it after the constants which is why `Half` does it that way.

Reply:
`Single` has it as private because its a special "built-in" type. `Half` is a "user defined type" and so needs it internal.

`Decimal32` is like `Half` in that regard.